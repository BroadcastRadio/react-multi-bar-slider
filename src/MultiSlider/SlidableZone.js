import PropTypes from 'prop-types';
import styled from '@emotion/styled';

const SlidableZone = styled('div')({
  position: 'absolute',
  width: '100%'
}, ({ size, zIndex }) => ({
  top: -size,
  bottom: -size,
  zIndex
}));

SlidableZone.propTypes = {
  size: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  zIndex: PropTypes.number.isRequired
};

export default SlidableZone;
