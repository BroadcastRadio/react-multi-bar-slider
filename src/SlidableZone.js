import PropTypes from 'prop-types';
import glamorous from 'glamorous';

const SlidableZone = glamorous.div({
  position: 'absolute',
  width: '100%'
}, ({ size, zIndex }) => ({
  top: -size,
  bottom: -size,
  zIndex
}));

SlidableZone.propTypes = {
  size: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  zIndex: PropTypes.number.isRequired
};

export default SlidableZone;
