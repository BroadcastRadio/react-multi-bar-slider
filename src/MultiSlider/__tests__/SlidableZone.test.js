import React from 'react';
import Enzyme, { render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import SlidableZone from '../SlidableZone';

Enzyme.configure({ adapter: new Adapter() });

beforeAll(() => {
  global.slidableZoneProps = {
    size: 40,
    zIndex: 5
  };
});

afterAll(() => {
  delete global.slidableZoneProps;
});

describe('SlidableZone.js', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(<SlidableZone {...slidableZoneProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const slidableZone = render(<SlidableZone {...slidableZoneProps} />);
    expect(slidableZone).toBeDefined();
  });
});
