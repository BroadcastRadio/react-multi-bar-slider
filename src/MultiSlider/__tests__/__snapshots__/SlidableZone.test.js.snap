// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlidableZone.js matches the snapshot 1`] = `
<div
  className="css-1r9s984"
  size={40}
/>
`;
